src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure implements AreaCalculator {
+
+    int bottomBase;
+    int topBase;
+    int side;
+
+    public IsoscelesTrapezoid(String color, String name, int bottomBase, int topBase, int side) {
+        super(color, name);
+        this.bottomBase = bottomBase;
+        this.topBase = topBase;
+        this.side = side;
+    }
+
+    @Override
+    public int calculateArea() {
+        return  ((bottomBase + topBase) / 2) * ((int)Math.sqrt((Math.pow(side, 2)) - Math.pow((((Math.pow((bottomBase - topBase), 2)) + (Math.pow(side, 2)) - (Math.pow(side, 2))) / (2 * (bottomBase - topBase))), 2)));
```

Comment:
The area calculation involves complex mathematical operations and integer division, which can lead to inaccuracies. Consider using `double` for more precision in calculations, especially when dealing with square roots and divisions.